Volley to the rescue

karllindmark opened the pull request:
@peter-budo 

Try it out, it's like magic!!!

Commit:
Volley to the rescue

Commit:
Find the relevant view (i.e. not on an activity level)

Commit:
Fix issues around Assignments (throwing Exceptions)

Commit:
Extracted empty style to styles.xml

Commit:
Touch up the stats fragments

Commit:
Fix crash when opening LoginActivity ++

Commit:
Add Subscription for Refresh event

Add subscription for refresh event via the menu

The need for duplicating the following code:

    @Subscribe
    public void onRefreshEvent(RefreshEvent event) {
        startLoadingData();
    }

is due to the fact that Otto currently doesn’t traverse the hierarchy
looking for the subscription methods. That means, that if we register a
child, then it will look for the subscription ONLY in the child, and
not the parent as well. Yuck.

Looks like it’s by design, due to performance:
https://github.com/square/otto/issues/26

karllindmark commented:
Addressing following Pivotal stories:

https://www.pivotaltracker.com/story/show/64426976 (Solve problems with ViewPager+Loader) :octocat: 
https://www.pivotaltracker.com/story/show/67599536 (java.lang.NullPointerException @ assignments) :octopus: 
https://www.pivotaltracker.com/story/show/67599528 (Assignments progress bar - was not related to the loading though :o: )

Possibly also partially addressing https://www.pivotaltracker.com/story/show/67416702 (Investigate + Refactor VehicleStatsFragment) by moving the work off the UI thread, but I'll leave it "open" as I know that you're looking into the work that its doing (independent of thread).

@peter-budo

karllindmark commented:
Build: http://ci.peterscorner.co.uk/job/BF4-Intel-PR/41/console

peter-budo commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/base/ui/BaseLoadingListFragment.java:
```diff
@@ -27,34 +33,46 @@ public void onCreate(final Bundle icicle) {
+
+    @Subscribe
+    public void onRefreshEvent(RefreshEvent event) {
+        Log.d("YOLO", getClass().getSimpleName() + " is refrehshing!");
```
Rubbish

peter-budo commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/activities/MainActivity.java:
```diff
@@ -82,6 +84,11 @@ public boolean onOptionsItemSelected(final MenuItem item) {
                 return true;
 
+            case R.id.ab_action_refresh:
+                Log.d("YOLO", "Refreshing...");
```
Rubbish

peter-budo commented:
1. Http Request needs to be replaced by Volley in licenses list
2. Why About tabs do not rotate where rest of app does?
3. Stats and Unlocks on rotation do not maintain last viewed tab, on rotation selection jumps to first tab
4. Changelog update

peter-budo commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/login/LoginActivity.java:
```diff
@@ -44,7 +40,7 @@
     private static final String RESET_PASSWORD_LINK = "https://signin.ea.com/p/web/resetPassword";
     private static final int GAME_ID_BF4 = 2048;
-    private static final int ID_LOADER_GET_SOLDIERS = 0;
+    private static final int ID_REQUEST_GET_SOLDIERS = 0;
```
Obsolete

peter-budo commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/awards/AwardGridFragment.java:
```diff
@@ -14,12 +13,16 @@
+import java.util.Set;
 
 public class AwardGridFragment extends BaseLoadingFragment {
     private static final int ID_LOADER = 1100;
```
Obsolete

peter-budo commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/assignments/AssignmentGridFragment.java:
```diff
@@ -38,56 +37,40 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
-    @Override
-    public void onResume() {
-        super.onResume();
+    @Subscribe
```
Obsolete loader di declaration line25

peter-budo commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/battlefeed/BattleFeedFragment.java:
```diff
@@ -54,9 +52,8 @@ public View onCreateView(final LayoutInflater inflater, final ViewGroup parent,
-    public void onResume() {
-        super.onResume();
+    @Subscribe
+    public void onRefreshEvent(RefreshEvent event) {
```
obsolete loader id

peter-budo commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/news/NewsArticleFragment.java:
```diff
@@ -95,80 +86,160 @@ protected void startLoadingData() {
 
             default:
-                throw new IllegalArgumentException("No loader matching " + loaderId);
+                throw new IllegalArgumentException("No request matching " + id);
```
I'm sure we already discussed this, but any chance of changing this to log instead of exception? No request is sent to queue so no harm in just login wrong request id

karllindmark replied:
Good point Peter. When using loaders, we had the issue where we either HAD to return something, or throw an exception. Here we can just log it. :+1: _fixes_

Commit:
Remove unwanted loggings

Commit:
Fix issue with multi-platform personas

Fixing issue with multi-platform personas, as reported by Mike. I feel
this is necessary to include in this PR as they should go hand in hand.

However, it’s ugly that they do share ID’s, so that we instead have to
rely on storing the positions instead. Bleh.

Link to Pivotal:
https://www.pivotaltracker.com/story/show/67605212

Commit:
Fix xml spacing

Commit:
Convert from throwing exception to logging

Commit:
Remove obsolete loader ID's

Commit:
Update changelog + licenses

Commit:
Active orientation change for AppInfoActivity

Commit:
Persist tab selection

karllindmark commented:
OK @peter-budo, anything left? :-)

Commit:
Fix issue with search displaying wrong platform

Commit:
Fix progressbar in Soldier Overview
